dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/ReadTimeoutTest.java

```diff
@@ -0,0 +1,229 @@
+     */
+    private static void setupldapServer() throws Exception {
+        ldapServer = new EmbeddedApacheDS("myLDAP");
+        ldapServer.addPartition("users", USER_BASE_DN);
+        ldapServer.addPartition("groups", GROUP_BASE_DN);
+        ldapServer.startServer();
+
+        /*
+         * Add the partition entries.
+         */
+        Entry entry = ldapServer.newEntry(USER_BASE_DN);
+        entry.add("objectclass", "organizationalunit");
+        entry.add("ou", "Test");
+        entry.add("ou", "TestUsers");
+        ldapServer.add(entry);
+
+        /*
+         * Create the user.
+         */
+        entry = ldapServer.newEntry(USER_DN);
+        entry.add("objectclass", "inetorgperson");
+        entry.add("uid", USER);
+        entry.add("sn", USER);
+        entry.add("cn", USER);
+        entry.add("userPassword", "password");
+        ldapServer.add(entry);
+    }
+
+    /**
+     * Convenience method to configure the Liberty server with an {@link LdapRegistry} configuration that
+     * will connect to {@link #ldapServer}.
+     *
+     * @throws Exception If there was an error configuring the server.
+     */
+    private static void updateLibertyServer() throws Exception {
+        ServerConfiguration server = emptyConfiguration.clone();
+
+        LdapRegistry ldap = new LdapRegistry();
+        ldap.setId("ldap1");
+        ldap.setRealm("LDAPRealm");
+        ldap.setHost("localhost");
+        ldap.setPort(String.valueOf(ldapServer.getLdapServer().getPort()));
+        ldap.setBaseDN(SUB_DN);
+        ldap.setBindDN(EmbeddedApacheDS.getBindDN());
+        ldap.setBindPassword(EmbeddedApacheDS.getBindPassword());
+        ldap.setLdapType("Custom");
+//        ContextPool cp = new ContextPool(true, 0, 2, 1, "2s", "5s");
+//        ldap.setContextPool(cp);
+//        SearchResultsCache src = new SearchResultsCache();
+//        src.setEnabled(false); // disable search cache so we can look up the same user over and over again
+//        ldap.setLdapCache(new LdapCache(null, src));
+
+        server.getLdapRegistries().add(ldap);
+        updateConfigDynamically(libertyServer, server);
+
+    }
+
+    @Test
+    public void testCustomReadTimeout() throws Exception {
+        updateLibertyServer();
+        ServerConfiguration config = libertyServer.getServerConfiguration();
+        LdapRegistry ldap = config.getLdapRegistries().get(0);
+
+        String findstr = ", iReadTimeout=null,";
+        List<String> errMsgs = libertyServer.findStringsInLogsAndTrace(findstr);
+        System.out.println(errMsgs);
+
+        findstr = ", iReadTimeout=60000,";
+        errMsgs = libertyServer.findStringsInLogsAndTrace(findstr);
+        assertFalse("Should have found, " + findstr, errMsgs.isEmpty());
+
+        ldap.setReadTimeout("3000");
+        updateConfigDynamically(libertyServer, config);
+
+        //calling the servlet resets the configuration
+        String returnUser = servlet.checkPassword(USER_DN, "password");
```

Comment:
Delete this. I don't think you even load the servlet into the server in the build. gradle, so I presume the servlet is not even running on the Liberty server.

```suggestion

```